src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -140,32 +153,54 @@ private async Task ApplyContext(HostAssignmentContext assignmentContext)
                 var filePath = Path.GetTempFileName();
                 await DownloadAsync(zipUri, filePath);
 
-                _logger.LogInformation($"Extracting files to '{options.ScriptPath}'");
-                ZipFile.ExtractToDirectory(filePath, options.ScriptPath, overwriteFiles: true);
-                _logger.LogInformation($"Zip extraction complete");
+                using (_metricsLogger.LatencyEvent(MetricEventNames.LinuxContainerSpecializationZipExtract))
+                {
+                    _logger.LogInformation($"Extracting files to '{options.ScriptPath}'");
+                    ZipFile.ExtractToDirectory(filePath, options.ScriptPath, overwriteFiles: true);
+                    _logger.LogInformation($"Zip extraction complete");
+                }
             }
         }
 
         private async Task DownloadAsync(Uri zipUri, string filePath)
         {
             var zipPath = $"{zipUri.Authority}{zipUri.AbsolutePath}";
-            _logger.LogInformation($"Downloading zip contents from '{zipPath}' to temp file '{filePath}'");
+            _logger.LogInformation($"Downloading zip contents from '{Utility.CleanUrl(zipPath)}' to temp file '{filePath}'");
 
-            var response = await _client.GetAsync(zipUri);
-            if (!response.IsSuccessStatusCode)
+            HttpResponseMessage response = null;
+
+            await Utility.InvokeWithRetriesAsync(async () =>
             {
-                string error = $"Error downloading zip content {zipPath}";
-                _logger.LogError(error);
-                throw new InvalidDataException(error);
-            }
+                using (_metricsLogger.LatencyEvent(MetricEventNames.LinuxContainerSpecializationZipDownload))
+                {
+                    try
+                    {
+                        var request = new HttpRequestMessage(HttpMethod.Get, zipUri);
+                        response = await _client.SendAsync(request);
+                        response.EnsureSuccessStatusCode();
+                    }
+                    catch (Exception e)
+                    {
+                        string error = $"Error downloading zip content {Utility.CleanUrl(zipPath)}";
+                        _logger.LogError(error);
+                        _logger.LogError(e.ToString());
+                        throw;
```

Comment:
Above you had the exception handling outside of the retry block, meaning you didn't have to rethrow. Why the difference here? I think the pattern you used on the other one is better.

Reply:
Agree. Fixing this.